hotfix release: v0.2.X minor enhancements

sidnym-ladrut opened the pull request:
- [x] add basic markdown support
- [x] disable "share" button for remote projects being viewed on personal ships
- [x] fix caching behavior for web files between releases (e.g. vA.B.C+1 will force reloads for all web files for a user with vA.B.C files)
- [x] allow simple config hot-swapping for users

sidnym-ladrut committed:
add fledgling markdown support ('proj-view' page only)

sidnym-ladrut committed:
add more principled markdown support (via udon)

sidnym-ladrut committed:
fix issue with 'proj-edit' introducing trailing whitespace on each save

sidnym-ladrut committed:
fix udon block and link rendering

sidnym-ladrut committed:
fix udon nested list rendering; improve interface uniformity of 'settings' button

sidnym-ladrut committed:
factor out 'hero-plaq' into a dedicated component

sidnym-ladrut committed:
use zero-md markdown renderer; upgrade to twind@1.1.3 (fixed shim bugs)

sidnym-ladrut committed:
revamp 'asset' endpoint to dynamic asset loading; move test configs into 'bare' desk

sidnym-ladrut committed:
add '-web-reload' thread to simplify configuration swapping workflows

sidnym-ladrut committed:
add '-config' thread to further simplify swapping workflows

fix #59: remove the share project button when viewing from non-source ship

sidnym-ladrut committed:
fix #59: remove share button on post-donate page for remote users also

sidnym-ladrut committed:
improve '-config' thread so it runs more quickly & reliably

sidnym-ladrut committed:
edit %tonic to refresh on any desk update (for automated JS/CSS edit reloads)

thelifeandtimes commented:
> allow simple config hot-swapping for users 

is the idea here to have a ui element for switching between mainnet ethereum and testnet sepolia on livenet urbit, and then also urbit fakeships? (if so, i very much like the first and am unopinionated about the latter)

sidnym-ladrut commented:
> > allow simple config hot-swapping for users
> 
> is the idea here to have a ui element for switching between mainnet ethereum and testnet sepolia on livenet urbit, and then also urbit fakeships? (if so, i very much like the first and am unopinionated about the latter)

This facility is primarily intended to allow users to trivially toggle debug messages on their ships, which will help us diagnose failures by looking at extended debug logs.

This feature could be extended to cover a "ethereum network switch" button, but I'm thinking that feature will be  integrated with MetaMask (e.g. you switch your MetaMask to Sepolia and it will highlight all the Sepolia projects you know about). This would mostly be useful for us, but this allows power users to live test/demo the app as well and it lays the groundwork for potential multi-chain support (or, at least, multi-network ethereum support).

sidnym-ladrut committed:
add real beta support for markdown rendering

move full desk recreation logic to 'meta/exec/regen' script
fix application order of rudder page adapters

sidnym-ladrut committed:
fix #77: fix signing issue with \r characters in project text

sidnym-ladrut committed:
make proposed stage publicly accessible + shareable

thelifeandtimes commented on desk/bare/web/fund/page/proj-edit.hoon:
```diff
@@ -103,12 +111,11 @@
                 ==
               ==
-              ;label(for "sum"): project description
+              ;label(for "sum"): project description (markdown)
```
Instead of adding this to the label, let's call out in the placeholder text that this field supports markdown.

thelifeandtimes commented on desk/bare/web/fund/page/proj-edit.hoon:
```diff
@@ -144,12 +151,11 @@
                       ==
                     ==
-                    ;label(for "m{<min>}s"): milestone description
+                    ;label(for "m{<min>}s"): milestone description (markdown)
```
same comment as with the project description label.

sidnym-ladrut committed:
move markdown field indicators from input labels to placeholder text

added a few minor 'zero-md' loading indicator experiments

sidnym-ladrut committed:
add reliable loading indicators to markdown elements

sidnym-ladrut committed:
v0.2.2 release commit

thelifeandtimes reviewed (approved):
LGTM
